qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/texture/presenter.py

```diff
@@ -0,0 +1,351 @@
+            else:
+                self.unassigned_wss.pop(self.unassigned_wss.index(ws))
+        self.redraw_table()
+
+    # ---- Internal data structure logic for tracking parameter table assignments -------
+
+    def ws_has_param(self, ws):
+        return ws in self.ws_assignments.keys()
+
+    def param_has_ws(self, param):
+        return param in self.param_assignments.keys()
+
+    def get_assigned_params(self):
+        return list(self.param_assignments.keys())
+
+    def unassign_params(self, params):
+        for param in params:
+            if self.param_has_ws(param):
+                # remove assignment between ws and param
+                assigned_ws = self.param_assignments.pop(param)
+                self.ws_assignments.pop(assigned_ws)
+                # add ws to the unassigned pile
+                self.unassigned_wss.append(assigned_ws)
+
+    def assign_unpaired_wss_and_params(self):
+        (self.unassigned_wss, self.unassigned_params, self.ws_assignments, self.param_assignments) = (
+            self.model.assign_unpaired_wss_and_params(
+                self.unassigned_wss, self.unassigned_params, self.ws_assignments, self.param_assignments
+            )
+        )
+
+    def all_wss_have_params(self):
+        return self.model.all_wss_have_params(*self.view.get_selected_workspaces())
+
+    def at_least_one_param_assigned(self):
+        return self.model.at_least_one_param_assigned(self.get_assigned_params())
+
+    def _has_selected_wss(self):
+        selected_wss, _ = self.view.get_selected_workspaces()
+        return self.model.has_selected_wss(selected_wss)
+
+    # ----- Table logic -----------------
+
+    def delete_selected_param_files(self):
+        selected_wss, selected_params = self.view.get_selected_workspaces()
+        self.unassign_params(selected_params)
+        self.redraw_table()
+
+    def select_all(self):
+        self.view.set_all_workspaces_selected(True)
+        self.set_crystal_inputs_enabled()
+
+    def deselect_all(self):
+        self.view.set_all_workspaces_selected(False)
+        self.set_crystal_inputs_enabled()
+
+    # --------- Xtal Logic -------------------------------
+
+    def on_set_crystal_clicked(self):
+        self.model.set_ws_xtal(
+            self.view.get_crystal_ws_prop(), self.view.get_lattice(), self.view.get_spacegroup(), self.view.get_basis(), self.view.get_cif()
+        )
+        self.redraw_table()
+
+    def on_set_all_crystal_clicked(self):
+        wss, _ = self.view.get_selected_workspaces()
+        self.model.set_all_ws_xtal(wss, self.view.get_lattice(), self.view.get_spacegroup(), self.view.get_basis(), self.view.get_cif())
+        self.redraw_table()
+
+    # --------- Pole Figure Logic --------------------------
+
+    def on_calc_pf_clicked(self):
+        wss, params = self.view.get_selected_workspaces()
+        # require at least one wss to be selected
+        if len(wss) > 0:
```

Comment:
I think here it's more pythonic to return early - I wouldn't mention it but for requesting other changes in this file!
```suggestion
        if len(wss) == 0:
            return
        # following line of code can be moved back an indent level
```